File: beecrowd/p1/PontosDeFeno.c
#include<stdio.h>
#include<string.h>

int main()
{
    int n1, n2;
    scanf("%d %d",&n1, &n2);
    char cargos[n1][50];
    int valores[n1];
    for(int i=0; i<n1; i++) {
        scanf("%s",cargos[i]);
        scanf("%d",&valores[i]);
    }
    for(int i=0; i<n2; i++) {
        int cont = 0;
        char palavra[30];
        while(1){
            scanf("%s",&palavra);
            if(strcmp(palavra, ".")==0) {
                printf("%d",cont);
                break;
            }
            for(int j=0; j<n1; j++) {
                if(strcmp(palavra, cargos[j])==0) {
                    cont += valores[j];
                }
            }
        }
    }
}

File: beecrowd/p1/Pontos_de_Feno.java
package beecrowd;

import java.util.ArrayList;
import java.util.Scanner;

public class Pontos_de_Feno {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int op = sc.nextInt(); // quantidade de opções de cargos que seriam avaliados
        int cargos = sc.nextInt(); // quantos textos seriam lidos

        ArrayList<String> nomes = new ArrayList<>(op);
        ArrayList<Integer> valores = new ArrayList<>(op);

       // aqui tá lendo os nomes dos cargos e os valores para cada
        for (int i = 0; i < op; i++) {
            String nome = sc.next();
            int valor = sc.nextInt();
            nomes.add(nome);
            valores.add(valor);
        }

        // aqui ta lendo os textos até chegar na variavel cargos
        int ponto = 0;
        while (ponto < cargos) {
            int contador = 0;
            
            while (sc.hasNext()) { // está sendo lida cada palavra
                String palavra = sc.next();
                if (palavra.equals(".")) {
                    ponto++;
                    break;
                } else {
                    // verifica se a palavra está na lista de nomes e acumula o valor
                    for (int i = 0; i < op; i++) {
                        if (nomes.get(i).equals(palavra)) {
                            contador += valores.get(i);
                        }
                    }
                }
            }
            // imprime a soma dos valores (salario deles)
            System.out.println(contador);
        }
        sc.close();
    }
}
